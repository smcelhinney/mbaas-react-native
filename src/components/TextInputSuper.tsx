import * as React from 'react';
import { Component } from 'react';

import { TextInput, Text, View } from 'react-native';

export default class TextInputSuper extends Component<any, any> {
  constructor(props: any) {
    super(props);

    this.state = {
      isFocused: false
    };
  }

  private shouldFocus() {
    this.setState({
      isFocused: true
    });
  }

  private shouldBlur() {
    this.props.onBlur();

    this.setState({
      isFocused: false
    });
  }

  public render() {
    const focusStyle = this.state.isFocused
      ? {
          backgroundColor: this.props.focusColor,
          shadowRadius: 9,
          shadowOpacity: 0.1,
          shadowColor: 'black',
          bottom: 0.5
        }
      : null;

    return (
      <TextInput
        {...this.props}
        style={[focusStyle, this.props.superInputStyles]}
        onFocus={() => this.shouldFocus()}
        onBlur={() => this.shouldBlur()}
        underlineColorAndroid="transparent"
      />
    );
  }
}
