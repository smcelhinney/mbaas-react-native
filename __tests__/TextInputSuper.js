import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import TextInputSuper from '../build/components/TextInputSuper.js';
import toJson from 'enzyme-to-json';

describe('Component: TextInputSuper', () => {

  it('renders correctly', () => {
    const wrapper = shallow(
      <TextInputSuper />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  })

})