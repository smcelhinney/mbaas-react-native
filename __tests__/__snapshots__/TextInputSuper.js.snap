// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: TextInputSuper renders correctly 1`] = `
<TextInput
  onBlur={[Function]}
  onFocus={[Function]}
  style={
    Array [
      null,
      undefined,
    ]
  }
  underlineColorAndroid="transparent"
/>
`;
